lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -488,6 +488,18 @@ def get_edgecolor(self):
         return self._maybe_depth_shade_and_sort_colors(super().get_edgecolor())
 
 
+def _get_data_scale(X, Y, Z):
+    """
+    Estimate the scale of the 3D data for use in depthshading
+    """
+    def _m(data):
+        if len(data) == 0:
```

Comment:
Aren't x, y, z, are the same size at this point, so you don't need separate checks for len(data)==0?

Reply:
Good point. Yes, I think they would all be the same size so only one check is needed.